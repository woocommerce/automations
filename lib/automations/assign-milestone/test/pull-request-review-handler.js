const { gimmeOctokit, gimmeContext, gimmeApp } = require( '@testHelpers' );

describe( 'pull-request-review-handler assigns milestones to pull requests', () => {
	let octokit, context, app;
	beforeEach( () => {
		octokit = gimmeOctokit();
		context = gimmeContext( 'pull_request_review', 'submitted' );
		app = gimmeApp( 'assign-milestone' );
	} );
	it( 'does nothing if the pull request is not approved', async () => {
		context.payload = {
			...context.payload,
			review: { state: 'comment' },
		};
		await app( context, octokit );
		expect( octokit.issues.update ).not.toHaveBeenCalled();
	} );
	it( 'does nothing if the pull request has a milestone', async () => {
		context.payload = {
			...context.payload,
			pull_request: { milestone: { number: 1 } },
		};
		await app( context, octokit );
		expect( octokit.issues.update ).not.toHaveBeenCalled();
	} );
	it( 'assigns a milestone to an approved pull request', async () => {
		await app( context, octokit );
		expect( octokit.repos.getContents ).toHaveBeenCalled();
		expect(
			octokit.issues.listMilestones.endpoint.merge
		).toHaveBeenCalled();
		expect( octokit.issues.update ).toHaveBeenCalled();
		expect( octokit.issues.update.mock.calls[ 0 ] ).toMatchSnapshot();
	} );
} );
