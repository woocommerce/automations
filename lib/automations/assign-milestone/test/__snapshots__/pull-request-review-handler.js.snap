// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pull-request-review-handler assigns milestones to pull requests assigns a milestone to an approved pull request 1`] = `
Array [
  Object {
    "issue_number": 40,
    "milestone": 1235,
    "owner": "nerrad",
    "repo": "tests",
  },
]
`;
